feature: new and improved thoughts system

Ryos00 opened the pull request:
- cats should no longer think about outsiders or dead cats when they're not supposed to
- thoughts are more organized

Ryos00 committed:
fix: returning lost cats now take into account exp level

Ryos00 committed:
Merge remote-tracking branch 'upstream/development' into dev

Ryos00 committed:
add: new thoughts stuff

Ryos00 committed:
change filename

Ryos00 committed:
add: more thoughts stuff

Ryos00 committed:
Merge remote-tracking branch 'upstream/development' into dev

Ryos00 committed:
update: add new constraints

Ryos00 committed:
Merge remote-tracking branch 'upstream/development' into dev

Ryos00 committed:
Merge branch 'keyraven-shaders' into development

CL0WNTH0UGHTS committed:
Update and Change warrior_to_other thoughts

-Updates the warrior to other thoughts to the new thought format
-adds 25 new thoughts for warriors to kits (alive, starclan, darkforest, and parent-to-child kits)

Ryos00 committed:
Merge remote-tracking branch 'upstream/development' into dev

Ryos00 committed:
Merge branch 'development' of https://github.com/CL0WNTH0UGHTS/clangen into CL0WNTH0UGHTS-development

Ryos00 committed:
Merge branch 'CL0WNTH0UGHTS-development' into dev

Ryos00 committed:
fix some things

Ryos00 committed:
update

Ryos00 committed:
Merge remote-tracking branch 'upstream/development' into dev

CL0WNTH0UGHTS committed:
reformat thoughts

no descriptions its 1 am

Ryos00 committed:
Merge pull request #87 from CL0WNTH0UGHTS/dev

reformat thoughts

Ryos00 committed:
Merge remote-tracking branch 'upstream/development' into dev

Ryos00 committed:
move some thoughts

Ryos00 committed:
fixing

Ryos00 committed:
add missing files as placeholders

Ryos00 committed:
idk what I'm doing

Ryos00 committed:
Key is my savior

Ryos00 committed:
fix: move thoughts stuff inside the thoughts class

Ryos00 committed:
Merge remote-tracking branch 'upstream/development' into dev

Ryos00 committed:
fix

Ryos00 committed:
FIX AGAIN

Ryos00 committed:
fixes: thoughts work!!

Ryos00 committed:
typo

Ryos00 committed:
Merge remote-tracking branch 'upstream/development' into dev

Ryos00 committed:
Merge remote-tracking branch 'upstream/development' into dev

Ryos00 committed:
Merge remote-tracking branch 'upstream/development' into dev

Ryos00 committed:
Merge remote-tracking branch 'upstream/development' into dev

Ryos00 committed:
fix

CL0WNTH0UGHTS committed:
delete the warrior traits .json

all trait thoughts have been converted and placed into the alive warrior.json

CL0WNTH0UGHTS committed:
Delete ALL trait.jsons

unneeded as all the traits have been converted and placed into their respective .jsons

CL0WNTH0UGHTS committed:
reformat every single other cat thoughts

while not the correct place, reformats all kittypet, loner, rogue, exiled, and lost thoughts

CL0WNTH0UGHTS committed:
re-move kitten.json

this asshole file created to many issues, had to move it to a new folder and is moving it back

CL0WNTH0UGHTS committed:
delete all of these files

includes all of the files in alive(old), family.json, and exile.json. unneeded as all the thoughts have been converted and placed in their proper places!

CL0WNTH0UGHTS committed:
Add all apprentice specifc thoughts to apprentice.json

Adds all medicine cat, mediator, and warrior apprentice thoughts into one place (includes traits!)

CL0WNTH0UGHTS committed:
reformts all deputy thoughts

Reformats and adds all deputy thoughts (traits included)

CL0WNTH0UGHTS committed:
reformat all general thoughts

reformats all general thoughts to the new format

CL0WNTH0UGHTS committed:
Update all leader thoughts

-Updates all leader thoughts (including traits) to the new format

CL0WNTH0UGHTS committed:
updates and removes mediator thoughts

-updates all mediator thoughts to the new format
-changed the location of (most) mediator apprentice thoughts

CL0WNTH0UGHTS committed:
Updates the dark forest thoughts

-reformats and updates all dark forest thoughts and places them in their correct place!

CL0WNTH0UGHTS committed:
update starclan thoughts

-reformats all starclan thoughts into the new format and place!

CL0WNTH0UGHTS committed:
update all UR thoughts

CL0WNTH0UGHTS committed:
updates all warrior trait thoughts

-reformats and replaces all warrior trait thoughts

CL0WNTH0UGHTS committed:
reformats, updates, and changes medicine cat.json

-formats and places all medicine cat specifc traits into the proper places
-yeets medicine cat apprentice thoughts to a new place

Ryos00 committed:
Merge pull request #88 from CL0WNTH0UGHTS/dev

Update all 1,600+ thoughts

Ryos00 committed:
backstory stuff

Ryos00 committed:
Merge branch 'dev' of https://github.com/Ryos00/clangen into dev

Ryos00 committed:
Merge remote-tracking branch 'upstream/development' into dev

Ryos00 committed:
move 'other' thoughts

Ryos00 committed:
moving + fixes

Ryos00 committed:
fixing things & adding new UR thoughts

Ryos00 committed:
fixes

Ryos00 committed:
some shuffling

Ryos00 committed:
add newborn thoughts

Ryos00 committed:
fix: cats won't think about long dead cats

Ryos00 committed:
fix: added an exception for dead cats

Ryos00 committed:
fix: outside cats weirdness

Ryos00 committed:
Merge remote-tracking branch 'upstream/development' into dev

Ryos00 committed:
fixes

keyraven committed:
Thoughts Changes

keyraven committed:
Merge branch 'dev' of https://github.com/Ryos00/clangen into dev

Ryos00 committed:
Merge pull request #90 from keyraven/dev

Thoughts Changes

Ryos00 committed:
little fix

Ryos00 committed:
fix: cats thinking about dead cats

Ryos00 committed:
add comment

Ryos00 committed:
lil fix

Ryos00 committed:
Merge remote-tracking branch 'upstream/development' into dev

Ryos00 committed:
idk how to fix tests

Ryos00 committed:
yeah

Ryos00 committed:
fixing import issue

Ryos00 committed:
add game mode to function

Lixxis commented on resources/dicts/thoughts/alive/alive_outside/exile.json:
```diff
@@ -0,0 +1,29 @@
+            "Misses their friends"
+        ],
+        "main_status_constraint": [
+            "exile"
```
```suggestion
            "exiled"
```

Lixxis commented on resources/dicts/thoughts/alive/alive_outside/general.json:
```diff
@@ -0,0 +1,66 @@
+            "Is missing r_c",
+            "Is scared that they will never be found"
+        ],
+        "main_status_constraint": [
```
Is the elder not allowed to think this?
The elder is missing in the status list - and if they are implemented like the relationship interactions, there is no need to list all status if everyone can think it.

Lixxis commented on resources/dicts/thoughts/alive/alive_outside/medicine cat.json:
Can we name it medicine_cat.json ? 
Then it would be in line with other naming conventions of the files (and sometimes spaces in file names, lead to problem when using command lines)

Lixxis commented on resources/dicts/thoughts/alive/deputy.json:
```diff
@@ -0,0 +1,69 @@
+        ],
+        "random_status_constraint": [
+            "deputy"
+        ]
```
Isn't there a 'live status' check?
I think there are missing some conditions, the main cat should also be a deputy, but the 'main_status_constraint' is missing

Lixxis commented on resources/dicts/thoughts/alive/deputy.json:
In general there is missing the 'main_status_constraint' not sure if this is intended.
If so, maybe also remove the constraint of the other files? Just to have a clear structure through all the thought json files.

Lixxis commented on resources/dicts/thoughts/alive/elder.json:
Same as alive/deputy json, 'main_status_constraint' is missing. 
If intended, pls remove it from all thoughts json.

Lixxis commented on resources/dicts/thoughts/alive/general.json:
Don't know if it is implemented as the interaction/relationship stuff, but if so there is no need to list all 'main_stauts_constraint'.
Please check if the list is really needed.

Lixxis commented on resources/dicts/thoughts/alive/kitten.json:
Same as alive/deputy and alive/elder json, 'main_status_constraint' is missing.
If intended, pls remove it from all thoughts json.

Lixxis commented on resources/dicts/thoughts/alive/leader.json:
Same as other alive/status json, 'main_status_constraint' is missing.
If intended, pls remove it from all thoughts json.

Lixxis commented on resources/dicts/thoughts/alive/mediator.json:
```diff
@@ -0,0 +1,238 @@
+            "Is proud of how well they are doing at mediating",
+            "Accidentally made an argument worse",
+            "Is proud of getting through a difficult mediation session"
+        ]
```
'main_status_constraint' is missing

Ryos00 committed:
fix: occasional crash

Lixxis commented on resources/dicts/thoughts/alive/medicine cat.json:
Can we name it medicine_cat.json ?
Then it would be in line with other naming conventions of the files (and sometimes spaces in file names, lead to problem when using command lines)

and 'main_status_constraint' is missing (see other comments I mentioned it)

Lixxis commented on resources/dicts/thoughts/alive/newborn.json:
```diff
@@ -0,0 +1,31 @@
+            "Snoozes peacefully",
+            "Tries to open their eyes, but it's all blurry"
+        ],
+        "main_age_constraint": [
```
Is it really main_age_constraint ? (can be)
But everywhere else the 'main_status_constraint' is used, maybe use this constraint to be more 'in line' with the overall structure?

Lixxis commented on resources/dicts/thoughts/alive/warrior.json:
Sometimes 'main_status_constraint' is missing (start at line 162).
See other comments like this.

Lixxis commented on resources/dicts/thoughts/dead/starclan/deputy.json:
```diff
@@ -0,0 +1,11 @@
+    {
+        "id": "deputy_dead_deputy",
+        "thoughts": [
+            "Wishes they could speak to r_c to learn how to assign patrols better"
```
A dead cat is thinking this? --> duplicate in another deputy thought (maybe check on that)

Lixxis commented on resources/dicts/thoughts/dead/starclan/deputy.json:
```diff
@@ -0,0 +1,11 @@
+        "thoughts": [
+            "Wishes they could speak to r_c to learn how to assign patrols better"
+        ],
+        "random_status_constraint": [
```
'main_status_constaint' is missing (see other comments)

Lixxis commented on resources/dicts/thoughts/dead/starclan/kitten.json:
```diff
@@ -0,0 +1,11 @@
+        ],
+        "random_status_constraint": [
+            "kitten"
+        ]
```
'main_status_constaint' is missing (see other comments)

Lixxis commented on resources/dicts/thoughts/dead/starclan/medicine cat.json:
```diff
@@ -0,0 +1,59 @@
+            "medicine Cat Apprentice"
+        ],
+        "random_living_status": [
+            "darkforest"
```
why is this thought than in the dead/**starclan**/medicine cat.json

Lixxis commented on resources/dicts/thoughts/dead/starclan/apprentice.json:
```diff
@@ -0,0 +1,23 @@
+        "thoughts": [
+            "Is wondering if r_c would have been their friend"
+        ],
+        "random_status_constraint": [
```
'main_status_constaint' is missing, yes age will also do, but for all other this constaint is used. Maybe also use this constraint to stay more 'in line' with the style.

Lixxis commented on resources/dicts/thoughts/dead/starclan/medicine cat.json:
Can we name it medicine_cat.json ?
Then it would be in line with other naming conventions of the files (and sometimes spaces in file names, lead to problem when using command lines)

Lixxis commented on resources/dicts/thoughts/dead/starclan/medicine cat.json:
```diff
@@ -0,0 +1,59 @@
+            "Kitten"
+        ],
+        "random_living_status": [
+            "darkforest"
```
why is this thought than in the dead/**starclan**/medicine cat.json

Lixxis commented on scripts/cat/cats.py:
```diff
@@ -33,10 +32,10 @@
 from scripts.cat_relations.relationship import Relationship
 from scripts.game_structure import image_cache
 from scripts.event_class import Single_Event
+from .thoughts import Thoughts
```
```suggestion
from scripts.cat.thoughts import Thoughts
```

Lixxis commented on scripts/cat/cats.py:
```diff
@@ -1030,23 +1029,27 @@ def thoughts(self):
         i = 0
         while other_cat == self.ID and len(all_cats) > 1 or (
-                all_cats.get(other_cat).status in ['kittypet', 'rogue', 'loner']):
+                all_cats.get(other_cat).status in ['kittypet', 'rogue', 'loner', 'former Clancat']):
```
Is 'former Clancat' a valid status? 
I thought 'exiled' is used for ex-Clancats.

Lixxis commented on scripts/cat/cats.py:
```diff
@@ -1030,23 +1029,27 @@ def thoughts(self):
             other_cat = random.choice(list(all_cats.keys()))
+            # makes sure that a cat won't think about a cat that was dead longer than they've been alive (with 4 moons difference)
+            if all_cats[other_cat].dead and not self.dead and not self.status in ['leader', 'medicine cat']:
+                if (int(all_cats[other_cat].dead_for) + 4) > self.moons:
```
I would change it to the way I told you with the relationship stuff.
This would fix the error with the old loner joining a clan fresh and suddenly know freshly died clan cats.

```suggestion
                if all_cats[other_cat].ID not in self.relationships:
```

Lixxis commented on scripts/events.py:
```diff
@@ -551,7 +551,7 @@ def handle_lost_cats_return(self):
         for cat in Cat.outside_cats.values():
             if cat.outside and cat.status not in [
-                    'kittypet', 'loner', 'rogue', 'former clancat'
+                    'kittypet', 'loner', 'rogue', 'former Clancat'
```
Is 'former Clancat' a valid status?
I thought 'exiled' is used for ex-Clancats.

Lixxis commented on scripts/events.py:
```diff
@@ -628,7 +628,7 @@ def one_moon_outside_cat(self, cat):
             cat.age = "kitten"
             if cat.status not in [
-                'kittypet', 'loner', 'rogue', 'former clancat'
+                'kittypet', 'loner', 'rogue', 'former Clancat'
```
Is 'former Clancat' a valid status?
I thought 'exiled' is used for ex-Clancats.

Lixxis commented on scripts/screens/cat_screens.py:
```diff
@@ -126,7 +126,7 @@ def bs_blurb_text(cat):
         return backstory_text.get(backstory, "")
     else:
-        if cat.status in ['kittypet', 'loner', 'rogue', 'former clancat']:
+        if cat.status in ['kittypet', 'loner', 'rogue', 'former Clancat']:
```
Is 'former Clancat' a valid status?
I thought 'exiled' is used for ex-Clancats.

Lixxis commented on scripts/screens/cat_screens.py:
```diff
@@ -915,7 +915,7 @@ def generate_column2(self, the_cat):
         # STATUS
         if the_cat.outside and not the_cat.exiled and not the_cat.status in ['kittypet', 'loner', 'rogue',
-                                                                             'former clancat']:
+                                                                             'former Clancat']:
```
Is 'former Clancat' a valid status?
I thought 'exiled' is used for ex-Clancats.

Lixxis commented on scripts/utility.py:
```diff
@@ -403,25 +403,25 @@ def create_outside_cat(Cat, status, backstory):
             name = choice(names.names_dict["loner_names"] +
                                  names.names_dict["normal_prefixes"])
-        elif status == 'former clancat':
+        elif status == 'former Clancat':
```
Is 'former Clancat' a valid status?
I thought 'exiled' is used for ex-Clancats.

Lixxis commented on resources/dicts/thoughts/dead/unknownresidence/medicine cat.json:
Can we name it medicine_cat.json ?
Then it would be in line with other naming conventions of the files (and sometimes spaces in file names, lead to problem when using command lines)

Lixxis commented on resources/dicts/thoughts/dead/darkforest/medicine cat.json:
Can we name it medicine_cat.json ?
Then it would be in line with other naming conventions of the files (and sometimes spaces in file names, lead to problem when using command lines)

Ryos00 committed:
fix: file names

Ryos00 committed:
fix: relationship thoughts

Ryos00 committed:
Merge remote-tracking branch 'upstream/development' into dev

Ryos00 committed:
fix

Ryos00 committed:
Merge remote-tracking branch 'upstream/development' into dev

CL0WNTH0UGHTS committed:
Creates a med_cat_app.json for the UR

This creates a empty medicine cat apprentice json file for the Unknown Residence folder

CL0WNTH0UGHTS committed:
Create a mediator_app.json file or UR

This creates a empty mediator apprentice json file for the Unknown Residence folder

CL0WNTH0UGHTS committed:
create med_cat_app.json file for SC

This creates a empty medicine cat apprentice json file for the Starclanfolder

CL0WNTH0UGHTS committed:
create a med_cat_app.json for DF

This creates a empty medicine cat apprentice json file for the Darkforest folder

CL0WNTH0UGHTS committed:
create a medi_app.json for starclan

This creates a empty mediator apprentice json file for the Darkforest folder

CL0WNTH0UGHTS committed:
create medi_app folder for DF

This creates a empty mediator apprentice json file for the Darkforest folder

CL0WNTH0UGHTS committed:
Create and move all solo medi_app thoughts

-Creates the mediator_apprentice.json for alive thoughts
-Moves every solo mediator apprentice thoughts to this json

CL0WNTH0UGHTS committed:
create and move every med_cat_app solo thought

-creates a medicine cat apprentice json file for the alive thoughts
-moves every solo medicine cat apprentice thought over

CL0WNTH0UGHTS committed:
Updates alive warrior.json

-Removes all instances of "main_status_constraint" as it was unneeded
-moved every altruistic, empathetic, and patient thought to compassionate as those three were removed

CL0WNTH0UGHTS committed:
Update alive apprentice.json

-Removes all instances of "main_status_constraint" as it was unneeded
-moved every altruistic, empathetic, and patient thought to compassionate as those three were removed

CL0WNTH0UGHTS committed:
Update alive general.json

-moved every instance where the main_status needed to be used here
-moved every altruistic, empathetic, and patient thought to compassionate as those three were removed

CL0WNTH0UGHTS committed:
Reformat alive kitten.json

-noticed a few formatting errors (where it was on the wrong line), those have been mended

CL0WNTH0UGHTS committed:
reformat alive leader.json

-like kitten.json, a few of the thoughts were on the wrong line and had to be bumped another

CL0WNTH0UGHTS committed:
Update alive mediator.json

-Removes all instances of "main_status_constraint" as it was unneeded
-moved every altruistic, empathetic, and patient thought to compassionate as those three were removed

CL0WNTH0UGHTS committed:
Update alive medicine cat.json

-Removes all instances of "main_status_constraint" as it was unneeded
-moved every altruistic, empathetic, and patient thought to compassionate as those three were removed

Ryos00 committed:
Merge pull request #91 from CL0WNTH0UGHTS/dev

Reformatting, few grammar, and moving around lots of thoughts

Ryos00 committed:
add: missing newborn thoughts (placeholders)

Ryos00 committed:
fix: backstory issue

Ryos00 committed:
status constraints

Ryos00 committed:
add: biome, camp, and season to thought constraints

Ryos00 committed:
fix: grammar

Ryos00 committed:
fix: season janked

Ryos00 committed:
fix: thoughts test

Ryos00 committed:
fix: current season

Lixxis commented on resources/dicts/thoughts/dead/darkforest/newborn.json:
```diff
@@ -0,0 +1,31 @@
+    {
+        "id": "gen_newborn_to_parent",
+        "thoughts":[
+            "Snuggles closer to r_c",
```
How can a dead newborn snuggle to their parents belly?

Overall these thoughts are for alive newborns and not dead  ones

Lixxis commented on resources/dicts/thoughts/dead/starclan/medicine_cat.json:
```diff
@@ -0,0 +1,59 @@
+    {
+        "id": "med_cat_dead_SC_gen",
+        "thoughts": [
+            "Wishes they could visit r_c in StarClan",
```
Isn't the main cat also in star clan or did I miss something?

General some thoughts follow this pattern.

Lixxis commented on resources/dicts/thoughts/dead/starclan/newborn.json:
```diff
@@ -0,0 +1,31 @@
+    {
+        "id": "gen_newborn_to_parent",
+        "thoughts":[
+            "Snuggles closer to r_c",
```
Is it on purpose, that the sarclan newborn is snuggling up to an alive cat?

Lixxis commented on resources/dicts/thoughts/dead/unknownresidence/newborn.json:
```diff
@@ -0,0 +1,31 @@
+    {
+        "id": "gen_newborn_to_parent",
+        "thoughts":[
+            "Snuggles closer to r_c",
```
Same as in starclan. Dead newborn snuggles to alive cat?

Ryos00 committed:
Update medicine_cat.json

Ryos00 committed:
Merge remote-tracking branch 'upstream/development' into dev

lillithrosepup commented on scripts/game_structure/load_cat.py:
```diff
@@ -282,7 +282,8 @@ def json_load():
         cat.children.extend(cat.faded_offspring)
-
+        
+        game.switches['error_message'] = f'There was an error when thoughts for cat #{cat} are created.'
```
why is this here? it looks like it would cause issued on loading cats?
